packages/cli/src/ui/components/FolderTrustDialog.test.tsx

```diff
@@ -21,16 +21,85 @@ describe('FolderTrustDialog', () => {
     );
   });
 
-  it('should call onSelect with DO_NOT_TRUST when escape is pressed', async () => {
+  it('should call onSelect with DO_NOT_TRUST when escape is pressed and not restarting', async () => {
     const onSelect = vi.fn();
     const { stdin } = renderWithProviders(
-      <FolderTrustDialog onSelect={onSelect} />,
+      <FolderTrustDialog
+        onSelect={onSelect}
+        onRestartRequest={vi.fn()}
+        isRestarting={false}
+      />,
     );
 
-    stdin.write('\x1b');
+    stdin.write('\x1b'); // escape key
 
     await waitFor(() => {
       expect(onSelect).toHaveBeenCalledWith(FolderTrustChoice.DO_NOT_TRUST);
     });
   });
+
+  it('should not call onSelect when escape is pressed and is restarting', async () => {
+    const onSelect = vi.fn();
+    const { stdin } = renderWithProviders(
+      <FolderTrustDialog
+        onSelect={onSelect}
+        onRestartRequest={vi.fn()}
+        isRestarting={true}
+      />,
+    );
+
+    stdin.write('\x1b'); // escape key
+
+    // Give it a moment to process, then assert that onSelect was NOT called.
+    await new Promise((resolve) => setTimeout(resolve, 100));
```

Comment:
don't timeout. use waitFor. same elsewhere in file.

Reply:
Done